### ISSUE TITLE
Bug in scan/spawn points loading in select_in_hex()

### ISSUE THREAD

**Issue by azotov** [RocketMap/RocketMap#1902]
## Expected Behavior

After initial band scan is completed restarting RocketMap should not lead to rescanning of fully scanned locations with done=True

## Current Behavior

After restarting RocketMap on really big map (steps=70) with initial scan completed, SpeedScan begins to do initial scan for a couple of rightmost columns again ignoring the fact that these locations did have initial scan completed in the previous run of  RocketMap.

## Possible Solution

My investigation showed that it was caused by incomplete loading of previously scanned locations by ScannedLocation.select_in_hex() where it was called from location_changed(). The missing cells were recreated by  ScannedLocation.new_loc(e[1]) with default values (done=False, band* = -1) and these default values were save in database by db_update_queue.put((ScannedLocation, initial)). So the data from the previous run of RocketMap was lost in database after program restart for some cells and initial scan was relaunched for them over and over again after each restart of RocketMap.

The same problem should exist also for spawnpoints loading as SpawnPoint.select_in_hex() is identical to ScannedLocation.select_in_hex().

The problem was solved by commenting out the code for clipping the diagonals of hexagon in select_in_hex() i.e. returning _all_ points selected from database without any clipping. It caused longer loading due to excessive all_scans.keys() and in_radius() calls for points outside of hexagon but none of points were missed.

So the bug in select_in_hex() is in the code which makes clipping of points which are inside the square limited by hex_bounds(), but outside of hexagon. The obvious problem in this code is in using distances for spherical earth model while _generate_locations() uses get_new_coords() which in turn uses geopy.distance.distance() which uses WGS84 ellipsoid earth model. The difference between two models are negligible for small distances but for greater distances these models give different results. So the code works correctly for small step sizes but start missing points when distances become rather big.
There may be another bug in the code which makes clipping of points as well - edges of sufficiently big hexagon on earth surface are not straight lines and the code does not take it into consideration.

I suggest to remove completely the code of hexagon clipping in both select_in_hex() functions. It is better to spend some computational time at the program start by precise selections of our points by all_scans.keys() and link_spawn_points() then to skip some points due to attempt of imperfect coarse clipping in select_in_hex().

## Steps to Reproduce (for bugs)

Make complete initial scan with speed-scan=true, step-limit:=70 and location in eastern hemisphere. Restart the RocketMap and the program ignores the easternmost scans data and starts initial scan for these points again.

**Comment by rocketrobot99** [comment]
Although I haven't tried running an -ST 70 hex (I am not sure my PC could handle it), my PR #1900  includes a change in the way ScannedLocation.select_in_hex() works. It uses the Cell_ids rather than Lat and Long locations. It might be worth a try as it includes many other minor fixes.

**Comment by azotov** [comment]
Yes, I like your change in SpawnPoint.select_in_hex(), it should fix the issue with spawnpoints and work faster without floating point distance calculations. I suggest you to add into PR #1900 a change in ScannedLocation.select_in_hex() as well to fix similar issue with scan locations:

```
    def select_in_hex(cls, center, steps):
        n, e, s, w = hex_bounds(center, steps)
        # Get all spawns in that box.
        sp = list(cls
                  .select()
                  .where((cls.latitude <= n) &
                         (cls.latitude >= s) &
                         (cls.longitude >= w) &
                         (cls.longitude <= e))
                  .dicts())
        return sp
```
